platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -0,0 +1,344 @@
+# TSS - Roster Proposal
+
+---
+
+## Summary
+
+This proposal outlines the design specification of rosters to support Threshold Signature Scheme (TSS) block signing.
+It details the roster's lifecycle, data storage, API, and necessary changes to associated components within the
+platform.
+
+| Metadata           | Entities                                      | 
+|--------------------|-----------------------------------------------|
+| Designers          | Kore, Anthony, Ed, Cody, Austin               |
+| Functional Impacts | TSS, Genesis, Cryptography, Reconnect         |
+| Related Proposals  | TSS-Library, TSS-Ledger-Id, TSS-Block-Signing |
+| HIPS               | N/A                                           |
+
+## Purpose and Context
+
+The introduction of the Threshold Signature Scheme (TSS) requires a new mechanism for managing node participation in
+consensus and block signing.
+Roster, an immutable subset of the address book, will provide this mechanism, ensuring efficient and secure key
+management for TSS operations.
+This proposal attempts to provide a specification for the behavior of rosters starting from their creation from the
+Candidate Address Book (CAB) to their terminal state within the TSS specification.
+A roster reaches a terminal state when it is either adopted by the platform or replaced by a new roster.
+
+The Future Address Book (FAB) is a dynamic list maintained within the App that reflects the desired future state of the
+network's nodes. It will be continuously updated by HAPI transactions, such as those that create, update, or delete
+nodes.
+The CAB is a snapshot of the Future Address Book (FAB), and it is created when the app decides to adopt a new address
+book.
+
+This proposal specifies that Roster(s) will be created from the CAB by the Hedera app and passed to the platform code.
+The mechanism for doing so is detailed below.
+
+### Requirements
+
+- Immutability: Rosters must be immutable to protect the integrity of its computed hash.
+- Efficient Storage: Roster data should be stored efficiently in the state to minimize overhead and unnecessary data
+  copying.
+- Clear API: A well-defined API should be provided for the app to create and submit candidate rosters, and for the
+  platform to manage the lifecycle of submitted rosters.
+- Components update: Platform components that currently rely on the use of the Address book (such as Reconnect, Network,
+  Event Validation etc.) must be adapted to use rosters instead.
+- Address book paradigm discontinued within the platform codebase, only use rosters instead
+
+### Architecture and/or Components
+
+###### Roster Lifecycle
+
+![](TSS%20Roster%20Lifecycle.drawio.svg)
```

Comment:
I tried to capture the fact the votes happen as spec'd by the TSS ledger ID proposal but i see the 'No vote' isn't shown to be there. I needed something to specify what happens to an unused Roster and why. I'll see how i can include that within the TSS ledger ID boundary.

Reply:
good catch. Updated. Thanks!

Reply:
What I was trying to say is that voting is not relevant in TSS-Roster proposal.  Voting is only relevant in the TSS-Ledger-ID proposal.   Voting happens in response to receiving enough TssMessages and a node says its ready to adopt the keys for the candidate roster.

Reply:
I understand that. However, a roster ends up either being adopted or abandoned. We need to represent this stage in the lifecycle diagram. The 'vote' here is just a label indicating the trigger for either of the end states. I think having that label helps, especially as it's (now) fully enclosed within the TSS ledger ID spec. What if i include a caption "The mechanism of the Yes vs No vote falls outside the scope of this diagram and belongs in the TSS ledger ID spec."?

Reply:
I didn't want to include adoption mechanics in the lifecycle diagram, since it's details for another proposal. I wanted to scope this lifecycle to just the start, transitions, and end state. One thing we can do is change the "Yes vote" label to "Yes vote, after software upgrade". Will that work?

Reply:
Gotcha. Will removing the labels work? Keep the same diagram but without the word `vote` in it? It is important to indicate the end states of rosters in a roster lifecycle diagram, whether that end state is being implemented in this design or otherwise. A lifecycle diagram can involve a journey of a thing from point A to point D, but a single design proposal can focus on the mechanics from A to B, another proposal from B to C and so on.